src/utils/cosmosdb_model_versioned.ts

```diff
@@ -131,7 +138,9 @@ export abstract class CosmosdbModelVersioned<
     const modelId = this.getModelId(o);
     return (currentVersion === undefined
       ? this.getNextVersion(this.getSearchKey(o))
-      : fromEither<CosmosErrors, NonNegativeInteger>(right(currentVersion))
+      : fromEither<CosmosErrors, NonNegativeInteger>(
+          right(incVersion(currentVersion))
+        )
```

Comment:
Can we use a more compact version?
```suggestion
      : taskEither.of<CosmosErrors, NonNegativeInteger>(
          incVersion(currentVersion)
        )
```